hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java

```diff
@@ -1595,8 +1595,12 @@ public void start() throws IOException {
 
     setStartTime();
 
-    // At this point leader is not known
-    scmHAMetricsUpdate(null);
+    if (SCMHAUtils.isSCMHAEnabled(configuration) && getScmHAManager().getRatisServer().getLeaderId() != null) {
+      scmHAMetricsUpdate(getScmHAManager().getRatisServer().getLeaderId().toString());
+    } else {
+      // At this point leader is not known
+      scmHAMetricsUpdate(null);
+    }
```

Comment:
Sorry, I didn't notice this earlier.  Please call `getLeaderId()` once, and use it in both the condition and to update metrics.

```suggestion
    RaftPeerId leaderId = SCMHAUtils.isSCMHAEnabled(configuration)
        ? getScmHAManager().getRatisServer().getLeaderId()
        : null;
    scmHAMetricsUpdate(Objects.toString(leaderId, null));
```